hatch.toml

```diff
@@ -25,6 +25,10 @@ overrides.matrix.deps.env-vars = [
 overrides.matrix.deps.pre-install-commands = [
     { if = [ "min" ], value = "uv run ci/scripts/min-deps.py pyproject.toml --all-extras -o ci/min-deps.txt" },
 ]
+# To prevent situations like https://github.com/pydata/xarray/issues/10419 going forward
+overrides.matrix.deps.post-install-commands = [
+    { if = [ "pre" ], value = "uv pip install git+https://github.com/pydata/xarray.git" },
+]
```

Comment:
Again, deeply hoping to remove all this xarray immediately after 0.12

Reply:
As said many times: don't do two unrelated `[uv] pip install`s into the same venv!

Reply:
Hmmm, let me try with a pre-install constraints file, good call.  Open to other ideas.